crates/rspack_binding_values/src/module.rs

```diff
@@ -209,6 +213,73 @@ impl ModuleDTO {
       .map(|block_id| DependenciesBlockDTO::new(block_id, self.compilation))
       .collect::<Vec<_>>()
   }
+
+  #[napi]
+  pub fn size(&self, ty: Either<String, ()>) -> f64 {
+    let module = self.module();
+    let ty = match ty {
+      Either::A(s) => Some(SourceType::from(s.as_str())),
+      Either::B(_) => None,
+    };
+    module.size(ty.as_ref(), self.compilation)
+  }
+}
+
+thread_local! {
+  pub static MODULE_INSTANCE_REFS: RefCell<HashMap<CompilationId, HashMap<Identifier, Ref>>> = Default::default();
```

Comment:
How are we going to collect these modules? Are we leaking them on purpose?

Reply:
Just avoid repeatedly create JS Class instances of Module too.

Reply:
It takes some time to review this PR. 

The biggest headache when I was reviewing this is the lifetimes of the corresponding napi objects cannot be easily tracked with thread local refs per se. 

This could be replaced with a more explicit implementation. But you can do it in future PRs.